x/auth/middleware/middleware.go

```diff
@@ -86,6 +86,8 @@ func NewDefaultTxHandler(options TxHandlerOptions) (tx.Handler, error) {
 		// Recover from panics. Panics outside of this middleware won't be
 		// caught, be careful!
 		RecoveryTxMiddleware,
+		// Branch out MultiStore in DeliverTx and commit if no error returned.
+		SnapshotTxMiddleware,
```

Comment:
I think we need to put his _above_ RecoveryTxMiddleware, right?

RecoveryTxMiddleware can return an error, in which case, we don't want to run `msCache.Write()`

Reply:
I'm not sure, isn't the `NewRunMsgsTxHandler` is the most inner while at the top?

Reply:
RunMsgsTxHandler is the innermost (it's a TxHandler, not a middleware)

But then, all others go from outer to inner:
https://github.com/cosmos/cosmos-sdk/blob/675be9d6dbef2d82dd1ca8ee790314b30b1bfe3a/x/auth/middleware/middleware.go#L15-L24

Reply:
I see, changed.

Reply:
Well, it's same as function composition `f . g = \x -> f( g(x) )`